### PR TITLE
chore: update err msg for policy verification

### PR DESCRIPTION
**Pull request by qweeah**
This PR rephrased error message returned during trust policy verification.

### CODE REVIEW

**Review by patrickzheng200 — COMMENTED** [review]
Have you checked the following?
1. run `go test -v .` under trustpolicy dir. All tests need to be passed.
2. run the notation CLI e2e test again, because there are test cases expecting this error message. Changing the error message would fail the e2e test.

**Comment by qweeah** [comment]
> 2\. Changing the error message would fail the e2e test. All e2e tests need to be passed as well.

Shouldn't that be done when bumping notation-go?

**Comment by patrickzheng200** [comment]
> > 2. Changing the error message would fail the e2e test. All e2e tests need to be passed as well.
> 
> Shouldn't that be done when bumping notation-go?

oh yeah, if you need to change the e2e, you could include the change in this PR: https://github.com/notaryproject/notation/pull/593 because this change is connected closely with your CLI trust policy command.
Merge the notation-go PR first -> go mod tidy on notation CLI side -> merge the notation CLI PR.

**Review by patrickzheng200 — DISMISSED** [review]
LGTM

File: verifier/trustpolicy/trustpolicy.go [hunk 1]
```diff
@@ -289,7 +289,7 @@ func LoadDocument() (*Document, error) {
 // SignatureVerification struct throws error if SignatureVerification is invalid
 func (signatureVerification *SignatureVerification) GetVerificationLevel() (*VerificationLevel, error) {
 	if signatureVerification.VerificationLevel == "" {
-		return nil, errors.New("trust policy statement is missing or has empty signature verification level, it must be specified")
+		return nil, errors.New("signature verification configuration is missing or contains empty verification level, it must be specified")
```
Comment (priteshbandi) [inline verifier/trustpolicy/trustpolicy.go:292]: This is first time I am seeing `signature verification configuration`, why do we want to move away from trust policy statement?

Comment (qweeah) [reply verifier/trustpolicy/trustpolicy.go:292]: The name comes from the corresponding field called `signatureVerification` in the [schema spec](https://github.com/notaryproject/notaryproject/blob/v1.0.0-rc.2/specs/trust-store-trust-policy.md#trust-policy-schema)

If we want to be accurate, how about changing the error message to `verification level cannot be empty`

Comment (qweeah) [reply verifier/trustpolicy/trustpolicy.go]: Why this change is needed: the validation error comes from the verification level, the original msg might make notation user think the statement itself is missing.

**Review by byronchien — APPROVED** [review]

**Review by patrickzheng200 — APPROVED** [review]
LGTM

**Review by priteshbandi — APPROVED** [review]
LGTM

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Comment by shizhMSFT** [comment]
I'm not able to merge this PR due to
image
caused by recent change by https://github.com/notaryproject/notation-core-go/pull/135

**Comment by shizhMSFT** [comment]
Updated the branch policy to match the workflow defined in `notation-core-go`.

**Comment by qweeah** [comment]
> @qweeah Have you checked the following?
> 
> 1. run `go test -v .` under trustpolicy dir. All tests need to be passed.
> 2. run the notation CLI e2e test again, because there are test cases expecting this error message. Changing the error message would fail the e2e test. All e2e tests need to be passed as well.

@patrickzheng200 I bumped notation-go including this change and ran an [E2E ](https://github.com/qweeah/notation/actions/runs/4636725254/jobs/8204886757) in my own fork, looks like no test will fail thus  no further change required on CLI side.

**Pull request merged**